generate_from_csv.py

```diff
@@ -359,71 +361,64 @@ def csv_to_dict(csv_path):
-        min_standing_time = 0
-
-    # filter for vehicle_type
-    vehicle_types = set(d['vehicle_type'] for d in input.values())
-    for vt in vehicle_types:
-        vt_line = {k: v for k, v in input.items() if v["vehicle_type"] == vt}
-        vehicle_number = 0
-        # sort list of vehicles by departure time
-        departures = {key: value for key, value in sorted(vt_line.items(),
-                      key=lambda x: x[1]['departure_time'])}
-        # sort list of vehicles by arrival time
-        arrivals = {key: value for key, value in sorted(vt_line.items(),
-                    key=lambda x: x[1]['arrival_time'])}
-        # get the first arrival
-        first_arrival_time = arrivals[list(arrivals.keys())[0]]["arrival_time"]
-        # parse through rotations in departure list and assign vehicle id
-        if min_standing_time == "None":
-            min_standing_time = 0
-        for rotation in departures.keys():
-            # as long as the first vehicle has not arrived yet: add vehicle_id number
-            if datetime.datetime.strptime(first_arrival_time, '%Y-%m-%d %H:%M:%S') + \
-                    datetime.timedelta(hours=min_standing_time) >= \
-                    datetime.datetime.strptime(vt_line[rotation]["departure_time"],
-                                               '%Y-%m-%d %H:%M:%S'):
-                vehicle_number += 1
-                input[rotation]["vehicle_id"] = vt + "_" + str(vehicle_number)
-            # as soon as departure is after the first arrival take first vehicle_number in line
-            elif datetime.datetime.strptime(arrivals[list(arrivals.keys())[0]]["arrival_time"],
-                                            '%Y-%m-%d %H:%M:%S') + \
-                    datetime.timedelta(hours=min_standing_time) < \
-                    datetime.datetime.strptime(vt_line[rotation]["departure_time"],
-                                               '%Y-%m-%d %H:%M:%S'):
-                arrival_rotation = list(arrivals.keys())[0]
-                a_bus_number = input[arrival_rotation]["vehicle_id"]
-                del arrivals[arrival_rotation]
-                input[rotation]["vehicle_id"] = a_bus_number
+    rotations_in_progress = []
+    idle_vehicles = []
+    vehicle_type_counts = {vehicle_type: 0 for vehicle_type in vehicle_types.keys()}
+
+    rotations = sorted(input, key=lambda d: d.get('departure_time'))
+
+    for rot in rotations:
+        # find vehicles that have completed rotation and stood for a minimum staning time
+        # mark those vehicle as idle
+        for r in rotations_in_progress:
+            # calculate min_standing_time deps
+            capacity = vehicle_types[r["vehicle_type"]]["capacity"]
+            cs_power = max([v[1] for v in vehicle_types[r["vehicle_type"]]['charging_curve']])
+            min_standing_time = (capacity / cs_power) * recharge_fraction
+
+            if datetime.datetime.strptime(rot["departure_time"], '%Y-%m-%d %H:%M:%S') > \
```

Comment:
1) repetition of datetime format -> make it a) variable (set in config?) or b) static (define format at beginning of file)
2) if-clause far too long to be readable. parse datetimes beforehand, give them nice variable names, also convert min_standing_time to timedelta:
```
departure_time = datetime.datetime.strptime(rot["departure_time"], DATETIME_FORMAT)
arrival_time = datetime.datetime.strptime(r["arrival_time"], DATETIME_FORMAT)
min_standing_time = datetime.timedelta(hours=min_standing_time)  # extract from loop

if departure_time - arrival_time > min_standing_time:
  ...
```